app/routines/transfer_authentications.rb

```diff
@@ -10,7 +10,12 @@ def exec(authentications, target_user)
     authentications = [authentications] if !(authentications.is_a? Array)
     authentications.each do |authentication|
       authentication_user = authentication.user
-      authentication.update_attribute(:user_id, target_user.id)
+      return outputs[:status] = :same_provider \
+            if Authentication.exists?(user_id: target_user.id, provider: authentication.provider)
```

Comment:
you probably added this because of my point (3), but I intended this pre-check to be right before `TransferAuthentications` was even called.